Description: install headers under DESTDIR

 The relatively new HHVM_INSTALL_HEADERS function has some special code to
 recursively copy files to the target, using file(COPY ). However, file(COPY)
 does not by default prefix with DESTDIR and would install the headers into the
 system paths, instead of under the supplied destination directory.
 .
 Prefix the target with $ENV{DESTDIR} to remedy this.
Author: Faidon Liambotis <paravoid@debian.org>
Forwarded: no
Last-Update: 2014-05-27

--- a/CMake/HPHPFunctions.cmake
+++ b/CMake/HPHPFunctions.cmake
@@ -179,7 +179,7 @@ function(HHVM_INSTALL_HEADERS TARGET SRC
         string(SUBSTRING ${dest_rel} 0 ${slash_pos} dest_rel)
       endif()
       file(COPY ${src_rel}
-        DESTINATION "${DEST}/${dest_rel}"
+        DESTINATION "$ENV{DESTDIR}${DEST}/${dest_rel}"
         NO_SOURCE_PERMISSIONS)
     endif()
   endforeach()
